src/microbe_stage/Microbe.cs

```diff
@@ -904,10 +904,25 @@ public void AIThink(float delta, Random random, MicrobeAICommonData data)
     /// </returns>
     public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompounds(CompoundCloudSystem clouds)
     {
+        var colonyMicrobes = Colony?.ColonyMembers ?? new List<Microbe> { this };
```

Comment:
The code should have a performance shortcut for non-colonies. So you should have a check if the Colony is null and then just do the simple logic and only if the colony is not null and this is the colony lead cell, then do the more complex operation.